### PR TITLE
Merge prost-fork, delete previous amino_rs code

### PR DESCRIPTION
**Pull request by Liamsi**
This should bring in the whole commit history of prost! (incl. the commits in my fork) and deletes the previous implementation, while still retaining our old history. 

Additionally, it renames the crate to prost-amino. I think it better describes what this aims to be (but I haven't thought long about this). Previously, it was called amino for our amino_rs implementation (or prost, for @danburkert's  implementation). 

We should update and shorten the Readme to explain the state of this repo, point to [go-amino](https://github.com/tendermint/go-amino) and to [prost](https://github.com/danburkert/prost), too.

I think circleci only fails because prost didn't run rustfmt as we do (on circelci). I want to deviate from prost as little as possible. Hence, I propose to turn of rustfmt here :-/ (removed via https://github.com/tendermint/amino_rs/pull/12/commits/7719a4950a5c3ea89542bf2c5ac17033769eb9eb) Additionally, it doesn't run clippy (removed via https://github.com/tendermint/amino_rs/pull/12/commits/4e4e910808e046b3f7e2750a00ab549fd4aa6999). Tests still fail on circelci (but pass locally). 

After this gets merged we need to update the Cargo.toml in the kms to point to this crate instead of my fork.

related to https://github.com/tendermint/kms/pull/38
closes https://github.com/tendermint/amino_rs/issues/11

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
+readme = "README.md"
+description = "A Protocol Buffers implementation for the Rust Language."
+keywords = ["protobuf", "serialization"]
+categories = ["encoding"]
 
+[workspace]
+members = [
+  "benchmarks",
+  "conformance",
+  "prost-build",
```
[the remark this answers is not in the document]
Comment (Liamsi) [reply Cargo.toml]: Hmm, I would only change names of sub-crates that I've modified (prost-derive and prost).

File: README.md [hunk 2]
```diff
@@ -1,9 +1,335 @@
-#Rust Amino
+[![Build Status](https://travis-ci.org/danburkert/prost.svg?branch=master)](https://travis-ci.org/danburkert/prost)
+[![Windows Build Status](https://ci.appveyor.com/api/projects/status/24rpba3x2vqe8lje/branch/master?svg=true)](https://ci.appveyor.com/project/danburkert/prost/branch/master)
+[![Documentation](https://docs.rs/prost/badge.svg)](https://docs.rs/prost/)
+[![Crate](https://img.shields.io/crates/v/prost.svg)](https://crates.io/crates/prost)
 
-Rust implementation of the Amino encoding standard.
+# *PROST!*
```
[the remark this answers is not in the document]
Comment (Liamsi) [reply README.md]: I'll address all other comments you've made (Thanks a lot for the review!) and provide a minimalistic Readme. If you have suggestions, or, want to write it, let me know.

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Liamsi) [reply Cargo.toml]: Pros: 
- better branding  
- signal that we deviate from the original crate here
Cons:
 - slightly more difficult to maintain changes from prost!

I think the single Con outweighs the Pros by far though.

**Comment by Liamsi** [comment]
Do you see any action items left on this @zmanian?

**Comment by zmanian** [comment]
@Liamsi merge if you think you are ready

**Comment by Liamsi** [comment]
There is still plenty of work to be done but I think it's reasonable to merge this. It is more complete to what we have in amino_rs now. I'll open a bunch issues for what still needs to be done.

**Pull request merged**